plugins/pipelines/create.js

```diff
@@ -113,6 +120,36 @@ module.exports = () => ({
 
                                                 return defaultCollection.update();
                                             })
+                                            .then(() => {
+                                                if (autoKeysGeneration) {
+                                                    return pipelineFactory.scm
+                                                        .addDeployKey({
+                                                            scmContext,
+                                                            checkoutUrl,
+                                                            token: pipelineToken
+                                                        })
+                                                        .then(privateDeployKey => {
+                                                            if (!privateDeployKey) {
```

Comment:
is this needed ? won't the `addDeployKey` promise reject ?

Reply:
Yeah a catch should have been used. I'll correct it. 
Apart from this, I guess I found another place where it is used wrongly implemented:
https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/pipelines/openPr.js#L71